### ISSUE TITLE
Split dealing with CSV from training and dataset cleaning in fit_dataset

### ISSUE THREAD

**Issue by KOLANICH** [titu1994/pyshac#9]
Hi again.

I'm implementing resumpltion and metaoptimization in [UniOpt](https://gitlab.com/KOLANICH/UniOpt.py), so for [pyshac backend](https://gitlab.com/KOLANICH/UniOpt.py/blob/master/UniOpt/backends/pyshac.py) I need an API to inject points into the optimizer from memory:
 a) bulk injection;
 b) individual point injection (needed for metaoptimization and should be efficient.

I guess `fit_dataset` does this, but
1. it deals with csv files, so I cannot use it as it is
2. it does too much work, so I don't want to recreate it

It would be better if the stuff worked not only with arrays, but also with iterators.

**Comment by titu1994** [comment]
I'm a bit confused as to what "API to inject points into the optimizer from memory" means. 

Do you wish to add samples to the engine without going through the fit dataset procedure ?

**Comment by KOLANICH** [comment]
I want to add points (hyperparams, loss) into the engine while it is operating.

Most of optimizers work in the "predict n points with the model -> evaluate function in those points ->  add the points and function values in these points to the model and get the model ready for the next iteration" loop

I want to add points to a model directly, without first 2 steps. This is meant to enable
* resumation. I just save all points and on resumation just inject them into an empty optimizer of any type for which injection is implemented.
* metaoptimization. I generate n points with each optimizer, evaluate the function in all of them, and then inject the points into each optimizer. Different optimizers are good on different tasks, or maybe even on different stages of optimization of the same task, so we may get best of all of them.
